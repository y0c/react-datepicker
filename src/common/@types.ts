export namespace IDatePicker {
  export type Locale = any;
  export enum PickerDirection {
    TOP,
    BOTTOM,
  }

  export enum ViewMode {
    YEAR,
    MONTH,
    DAY,
  }

  export enum TimeType {
    AM = 'AM',
    PM = 'PM',
  }
  export interface Position {
    left?: string;
    top?: string;
    bottom?: string;
    right?: string;
  }

  export interface SVGIconProps {
    size?: string;
    className?: string;
    color?: string;
    onClick?: () => void;
    style?: {};
  }
}
