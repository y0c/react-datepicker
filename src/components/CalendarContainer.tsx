import * as classNames from 'classnames';
import * as dayjs from 'dayjs';
import * as React from 'react';
import { IDatePicker } from '../common/@types';
import CalendarBody from './CalendarBody';
import CalendarHead, { HeaderButtonProps } from './CalendarHead';
import { Props as DayViewProps } from './DayView';
import TodayPanel from './TodayPanel';
import { ifExistCall } from '../utils/FunctionUtil';
import { DatePickerDefaults } from '../common/Constant';
import { getToday } from '../utils/LocaleUtil';

interface CalendarContainerProps {
  /** Locale to use */
  locale?: IDatePicker.Locale;
  /** Calendar Show or Hide */
  show?: boolean;
  /** PrevIcon Show or Hide */
  prevIcon?: ((props: HeaderButtonProps) => JSX.Element) | boolean;
  /** NextIcon Show or Hide */
  nextIcon?: ((props: HeaderButtonProps) => JSX.Element) | boolean;
  /** Event for Calendar day click */
  onChange?: (date: dayjs.Dayjs) => void;
  /** TodayPanel show or hide */
  showToday?: boolean;
  /** Format header title */
  formatHeaderTitle?: (viewMode: IDatePicker.ViewMode, current: dayjs.Dayjs) => string;
}

interface PrivateProps {
  /** CalendarContainer base prop */
  current: dayjs.Dayjs;
  /** Default Date parameter in calendar, which is the parent component */
  base: dayjs.Dayjs;
  /** Number of months to show at once */
  showMonthCnt: number;
  /** Set Calendar initial Date  */
  setBase: (base: dayjs.Dayjs) => void;
}

export interface State {
  viewMode: IDatePicker.ViewMode;
}

export type InheritProps = DayViewProps & CalendarContainerProps;
export type Props = CalendarContainerProps & DayViewProps & PrivateProps;

class CalendarContainer extends React.Component<Props, State> {
  public static defaultProps = {
    current: dayjs(),
    show: true,
    showMonthCnt: 1,
    showToday: false,
    locale: DatePickerDefaults.locale,
  };

  public state = {
    viewMode: IDatePicker.ViewMode.DAY,
  };

  constructor(props: Props) {
    super(props);
  }

  public getHeaderTitle = () => {
    const { current, formatHeaderTitle } = this.props;

    if (formatHeaderTitle) {
      return formatHeaderTitle(this.state.viewMode, current);
    }

    const year = dayjs(current).year();
    return {
      [IDatePicker.ViewMode.YEAR]: `${year - 4} - ${year + 5}`,
      [IDatePicker.ViewMode.MONTH]: `${year}`,
      [IDatePicker.ViewMode.DAY]: dayjs(current).format('YYYY.MM'),
    }[this.state.viewMode];
  };

  public handleTitleClick = () => {
    const { viewMode } = this.state;
    const { showMonthCnt } = this.props;
    let changedMode: IDatePicker.ViewMode = viewMode;

    if (viewMode === IDatePicker.ViewMode.MONTH) {
      changedMode = IDatePicker.ViewMode.YEAR;
    } else if (viewMode === IDatePicker.ViewMode.DAY) {
      changedMode = IDatePicker.ViewMode.MONTH;
    }
    this.setState({
      viewMode: showMonthCnt > 1 ? IDatePicker.ViewMode.DAY : changedMode,
    });
  };

  public handleChange = (value: string) => {
    const { viewMode } = this.state;
    const { current, onChange, setBase, showMonthCnt, base } = this.props;
    if (!value.trim()) return;
    if (showMonthCnt > 1) {
      const date = dayjs(current)
        .date(parseInt(value, 10))
        .toDate();
      ifExistCall(onChange, date);
      return;
    }

    if (viewMode === IDatePicker.ViewMode.YEAR) {
      setBase(dayjs(base).year(parseInt(value, 10)));
      this.setState({
        viewMode: IDatePicker.ViewMode.MONTH,
      });
    } else if (viewMode === IDatePicker.ViewMode.MONTH) {
      setBase(dayjs(base).month(parseInt(value, 10)));
      this.setState({
        viewMode: IDatePicker.ViewMode.DAY,
      });
    } else {
      const date = dayjs(current).date(parseInt(value, 10));
      ifExistCall(onChange, date);
    }
  };

  public handleBase = (method: string) => () => {
    const { base, setBase } = this.props;
    const { viewMode } = this.state;
    const date = dayjs(base);
    if (viewMode === IDatePicker.ViewMode.YEAR) {
      setBase(date[method](10, 'year'));
    } else if (viewMode === IDatePicker.ViewMode.MONTH) {
      setBase(date[method](1, 'year'));
    } else {
      setBase(date[method](1, 'month'));
    }
  };

  public handleToday = () => {
    const { setBase } = this.props;
    setBase(dayjs());
  };

  public renderCalendarHead = () => {
    const { prevIcon, nextIcon } = this.props;
    return (
      <CalendarHead
        onPrev={this.handleBase('subtract')}
        onNext={this.handleBase('add')}
        prevIcon={prevIcon}
        nextIcon={nextIcon}
        onTitleClick={this.handleTitleClick}
        title={this.getHeaderTitle()}
      />
    );
  };

  public renderTodayPane = () => {
    const { showToday, locale = DatePickerDefaults.locale } = this.props;
    return <TodayPanel today={getToday(locale)} onClick={this.handleToday} show={showToday} />;
  };

  public renderCalendarBody = () => {
    const {
      customDayClass,
      customDayText,
      disableDay,
      selected,
      startDay,
      endDay,
      onMouseOver,
      current,
      locale = DatePickerDefaults.locale,
    } = this.props;

    return (
      <CalendarBody
        viewMode={this.state.viewMode}
        current={current}
        selected={selected}
        startDay={startDay}
        endDay={endDay}
        disableDay={disableDay}
        onClick={this.handleChange}
        onMouseOver={onMouseOver}
        customDayClass={customDayClass}
        customDayText={customDayText}
        locale={locale}
      />
    );
  };

  public render() {
    const { show, showToday } = this.props;
    const calendarClass = classNames('calendar__container', {
      'calendar--show': show,
    });

    return (
      <div className={calendarClass}>
        {this.renderCalendarHead()}
        {showToday && this.renderTodayPane()}
        {this.renderCalendarBody()}
      </div>
    );
  }
}

export default CalendarContainer;
