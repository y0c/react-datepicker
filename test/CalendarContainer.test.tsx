import { mount, shallow, ShallowWrapper, ReactWrapper } from 'enzyme';

import * as dayjs from 'dayjs';
import * as React from 'react';
import * as sinon from 'sinon';

import 'dayjs/locale/en';
import 'dayjs/locale/ko';

import CalendarContainer, { Props, State } from '../src/components/CalendarContainer';
import { IDatePicker } from '../src/common/@types';

describe('<CalendarContainer/>', () => {
  const current = dayjs(new Date(2018, 11, 5));
  let base = current;
  const setBase = (val: dayjs.Dayjs) => {
    base = val;
  };
  const defaultProps = {
    base,
    setBase,
    current,
  };

  let shallowComponent: ShallowWrapper<React.Component<Props>>;
  let mountComponent: ReactWrapper<Props, State>;

  describe('prop: show', () => {
    beforeEach(() => {
      shallowComponent = shallow(<CalendarContainer {...defaultProps} show />);
    });

    it('should render correctly', () => {
      expect(shallowComponent).toBeTruthy();
      expect(shallowComponent).toMatchSnapshot();
    });

    it('props show correctly', () => {
      expect(shallowComponent.hasClass('calendar--show')).toBeTruthy();
    });
  });

  describe('prop: locale', () => {
    let en: ReactWrapper<Props>;
    let ko: ReactWrapper<Props>;

    beforeEach(() => {
      en = mount(<CalendarContainer {...defaultProps} locale="en" />);
      ko = mount(<CalendarContainer {...defaultProps} locale="ko" />);
    });

    it('should locale prop correctly', () => {
      expect(ko).toMatchSnapshot();
      expect(
        en
          .find('th')
          .first()
          .text()
      ).toEqual('Sun');
      expect(
        ko
          .find('th')
          .first()
          .text()
      ).toEqual('일');
    });
  });

  describe('prop: monthNameOnHeader', () => {
    let component: ReactWrapper<Props>;
    let componentNameOnHeader: ReactWrapper<Props>;

    beforeEach(() => {
      component = mount(
        <CalendarContainer {...defaultProps} />
      );
      componentNameOnHeader = mount(
        <CalendarContainer {...defaultProps} monthNameOnHeader />
      );
    });

    it('should not change format of calendar header', () => {
      expect(
        component.find('.calendar__head--title').first().text()
      ).toEqual('2018.12');
    });

    it('should change format of calendar header', () => {
      expect(
        componentNameOnHeader.find('.calendar__head--title').first().text()
      ).toEqual('Dec, 2018');
    });
  });

  describe('handle base test(onPrev, onNext)', () => {
    beforeEach(() => {
      mountComponent = mount(
        <CalendarContainer {...defaultProps} prevIcon nextIcon showMonthCnt={1} />
      );
    });

    it('should onPrev correctly', () => {
      mountComponent.setState({
        viewMode: IDatePicker.ViewMode.DAY,
      });
      mountComponent.find('.calendar__head--prev > button').simulate('click');
      expect(dayjs(base).format('MM')).toEqual('11');

      mountComponent.setState({
        viewMode: IDatePicker.ViewMode.MONTH,
      });
      mountComponent.find('.calendar__head--prev > button').simulate('click');
      expect(dayjs(base).format('YYYY')).toEqual('2017');

      mountComponent.setState({
        viewMode: IDatePicker.ViewMode.YEAR,
      });
      mountComponent.find('.calendar__head--prev > button').simulate('click');
      expect(dayjs(base).format('YYYY')).toEqual('2008');
    });
  });

  describe('handle today test', () => {
    it('should handle today correctly', () => {
      mountComponent = mount(<CalendarContainer {...defaultProps} showToday prevIcon nextIcon />);
      mountComponent.setState({
        viewMode: IDatePicker.ViewMode.DAY,
      });
      mountComponent.find('.calendar__head--prev > button').simulate('click');
      mountComponent.find('.calendar__head--prev > button').simulate('click');
      mountComponent.find('.calendar__panel--today h2').simulate('click');
      expect(dayjs(base).format('YYYYMMDD')).toEqual(dayjs().format('YYYYMMDD'));
    });
  });

  describe('handle title click test', () => {
    it('should viewMode change correctly', () => {
      mountComponent = mount(<CalendarContainer {...defaultProps} />);
      // once click expect month
      mountComponent.find('.calendar__head--title').simulate('click');
      expect(mountComponent.state().viewMode).toEqual(IDatePicker.ViewMode.MONTH);

      // twice click expect year
      mountComponent.find('.calendar__head--title').simulate('click');
      expect(mountComponent.state().viewMode).toEqual(IDatePicker.ViewMode.YEAR);
    });

    it('should showMontCnt > 1 viewMode noChange', () => {
      mountComponent = mount(<CalendarContainer {...defaultProps} showMonthCnt={2} />);
      // once click expect day(showMontCnt > 1)
      mountComponent.find('.calendar__head--title').simulate('click');
      expect(mountComponent.state().viewMode).toEqual(IDatePicker.ViewMode.DAY);
    });
  });

  describe('handle change test', () => {
    it('should showMontCnt > 1 onChange call', () => {
      const onChange = sinon.spy();
      mountComponent = mount(
        <CalendarContainer
          {...defaultProps}
          base={dayjs(new Date(2018, 12, 4))}
          showMonthCnt={2}
          onChange={onChange}
        />
      );

      // empty date click
      mountComponent
        .find('td')
        .at(1)
        .simulate('click');

      expect(onChange).toHaveProperty('callCount', 0);

      mountComponent
        .find('td')
        .at(6)
        .simulate('click');

      expect(onChange).toHaveProperty('callCount', 1);
    });

    describe('Mode Specific test', () => {
      let onChange: sinon.SinonSpy;
      beforeEach(() => {
        onChange = sinon.spy();
        mountComponent = mount(<CalendarContainer {...defaultProps} onChange={onChange} />);
      });

      it('should year mode test', () => {
        mountComponent.setState({
          viewMode: IDatePicker.ViewMode.YEAR,
        });

        mountComponent
          .find('td')
          .at(6)
          .simulate('click');

        expect(dayjs(base).format('YYYY')).toEqual('2020');
        expect(mountComponent.state().viewMode).toEqual(IDatePicker.ViewMode.MONTH);
      });

      it('should month mode test', () => {
        mountComponent.setState({
          viewMode: IDatePicker.ViewMode.MONTH,
        });
        mountComponent
          .find('td')
          .at(1)
          .simulate('click');

        expect(dayjs(base).format('MM')).toEqual('02');
        expect(mountComponent.state().viewMode).toEqual(IDatePicker.ViewMode.DAY);
      });

      it('should day mode test', () => {
        mountComponent.setState({
          viewMode: IDatePicker.ViewMode.DAY,
        });

        mountComponent
          .find('td')
          .at(6)
          .simulate('click');

        expect(onChange).toHaveProperty('callCount', 1);
      });
    });
  });
});
