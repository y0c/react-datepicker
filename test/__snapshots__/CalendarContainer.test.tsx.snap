// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarContainer/> prop: locale should locale prop correctly 1`] = `
<CalendarContainer
  base={"2018-12-04T15:00:00.000Z"}
  current={"2018-12-04T15:00:00.000Z"}
  locale="ko"
  setBase={[Function]}
  show={true}
  showMonthCnt={1}
  showToday={false}
>
  <div
    className="calendar__container calendar--show"
  >
    <CalendarHead
      onNext={[Function]}
      onPrev={[Function]}
      onTitleClick={[Function]}
      title="2018.12"
    >
      <div
        className="calendar__head"
      >
        <div
          className="calendar__head--prev"
        />
        <h2
          className="calendar__head--title"
          onClick={[Function]}
        >
          2018.12
        </h2>
        <div
          className="calendar__head--next"
        />
      </div>
    </CalendarHead>
    <CalendarBody
      current={"2018-12-04T15:00:00.000Z"}
      locale="ko"
      onClick={[Function]}
      viewMode={2}
    >
      <div
        className="calendar__body"
      >
        <DayView
          current={"2018-12-04T15:00:00.000Z"}
          locale="ko"
          onClick={[Function]}
          viewMode={2}
        >
          <TableMatrixView
            cell={[Function]}
            headers={
              Array [
                "일",
                "월",
                "화",
                "수",
                "목",
                "금",
                "토",
              ]
            }
            matrix={
              Array [
                Array [
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                  "1",
                ],
                Array [
                  "2",
                  "3",
                  "4",
                  "5",
                  "6",
                  "7",
                  "8",
                ],
                Array [
                  "9",
                  "10",
                  "11",
                  "12",
                  "13",
                  "14",
                  "15",
                ],
                Array [
                  "16",
                  "17",
                  "18",
                  "19",
                  "20",
                  "21",
                  "22",
                ],
                Array [
                  "23",
                  "24",
                  "25",
                  "26",
                  "27",
                  "28",
                  "29",
                ],
                Array [
                  "30",
                  "31",
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                ],
              ]
            }
          >
            <table
              className="calendar__body--table"
            >
              <thead>
                <tr>
                  <th
                    key="0"
                  >
                    일
                  </th>
                  <th
                    key="1"
                  >
                    월
                  </th>
                  <th
                    key="2"
                  >
                    화
                  </th>
                  <th
                    key="3"
                  >
                    수
                  </th>
                  <th
                    key="4"
                  >
                    목
                  </th>
                  <th
                    key="5"
                  >
                    금
                  </th>
                  <th
                    key="6"
                  >
                    토
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  key="0"
                >
                  <Cell
                    className=""
                    key="0"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="1"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="2"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="3"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="4"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="5"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--6"
                    key="6"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="1"
                  >
                    <td
                      className="calendar__day calendar__day--6"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        1
                      </div>
                    </td>
                  </Cell>
                </tr>
                <tr
                  key="1"
                >
                  <Cell
                    className="calendar__day calendar__day--0"
                    key="7"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="2"
                  >
                    <td
                      className="calendar__day calendar__day--0"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        2
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--1"
                    key="8"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="3"
                  >
                    <td
                      className="calendar__day calendar__day--1"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        3
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--2"
                    key="9"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="4"
                  >
                    <td
                      className="calendar__day calendar__day--2"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        4
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--3"
                    key="10"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="5"
                  >
                    <td
                      className="calendar__day calendar__day--3"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        5
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--4"
                    key="11"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="6"
                  >
                    <td
                      className="calendar__day calendar__day--4"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        6
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--5"
                    key="12"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="7"
                  >
                    <td
                      className="calendar__day calendar__day--5"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        7
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--6"
                    key="13"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="8"
                  >
                    <td
                      className="calendar__day calendar__day--6"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        8
                      </div>
                    </td>
                  </Cell>
                </tr>
                <tr
                  key="2"
                >
                  <Cell
                    className="calendar__day calendar__day--0"
                    key="14"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="9"
                  >
                    <td
                      className="calendar__day calendar__day--0"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        9
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--1"
                    key="15"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="10"
                  >
                    <td
                      className="calendar__day calendar__day--1"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        10
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--2"
                    key="16"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="11"
                  >
                    <td
                      className="calendar__day calendar__day--2"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        11
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--3"
                    key="17"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="12"
                  >
                    <td
                      className="calendar__day calendar__day--3"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        12
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--4"
                    key="18"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="13"
                  >
                    <td
                      className="calendar__day calendar__day--4"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        13
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--5"
                    key="19"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="14"
                  >
                    <td
                      className="calendar__day calendar__day--5"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        14
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--6"
                    key="20"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="15"
                  >
                    <td
                      className="calendar__day calendar__day--6"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        15
                      </div>
                    </td>
                  </Cell>
                </tr>
                <tr
                  key="3"
                >
                  <Cell
                    className="calendar__day calendar__day--0"
                    key="21"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="16"
                  >
                    <td
                      className="calendar__day calendar__day--0"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        16
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--1"
                    key="22"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="17"
                  >
                    <td
                      className="calendar__day calendar__day--1"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        17
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--2"
                    key="23"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="18"
                  >
                    <td
                      className="calendar__day calendar__day--2"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        18
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--3"
                    key="24"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="19"
                  >
                    <td
                      className="calendar__day calendar__day--3"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        19
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--4"
                    key="25"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="20"
                  >
                    <td
                      className="calendar__day calendar__day--4"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        20
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--5"
                    key="26"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="21"
                  >
                    <td
                      className="calendar__day calendar__day--5"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        21
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--6"
                    key="27"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="22"
                  >
                    <td
                      className="calendar__day calendar__day--6"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        22
                      </div>
                    </td>
                  </Cell>
                </tr>
                <tr
                  key="4"
                >
                  <Cell
                    className="calendar__day calendar__day--0"
                    key="28"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="23"
                  >
                    <td
                      className="calendar__day calendar__day--0"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        23
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--1"
                    key="29"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="24"
                  >
                    <td
                      className="calendar__day calendar__day--1"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        24
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--2"
                    key="30"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="25"
                  >
                    <td
                      className="calendar__day calendar__day--2"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        25
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--3"
                    key="31"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="26"
                  >
                    <td
                      className="calendar__day calendar__day--3"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        26
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--4"
                    key="32"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="27"
                  >
                    <td
                      className="calendar__day calendar__day--4"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        27
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--5"
                    key="33"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="28"
                  >
                    <td
                      className="calendar__day calendar__day--5"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        28
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--6"
                    key="34"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="29"
                  >
                    <td
                      className="calendar__day calendar__day--6"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        29
                      </div>
                    </td>
                  </Cell>
                </tr>
                <tr
                  key="5"
                >
                  <Cell
                    className="calendar__day calendar__day--0"
                    key="35"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="30"
                  >
                    <td
                      className="calendar__day calendar__day--0"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        30
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className="calendar__day calendar__day--1"
                    key="36"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text="31"
                  >
                    <td
                      className="calendar__day calendar__day--1"
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                        31
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="37"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="38"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="39"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="40"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                  <Cell
                    className=""
                    key="41"
                    onClick={[Function]}
                    onMouseOver={[Function]}
                    subText=""
                    text=" "
                  >
                    <td
                      className=""
                      onClick={[Function]}
                      onMouseOver={[Function]}
                    >
                      <div>
                         
                      </div>
                    </td>
                  </Cell>
                </tr>
              </tbody>
            </table>
          </TableMatrixView>
        </DayView>
      </div>
    </CalendarBody>
  </div>
</CalendarContainer>
`;

exports[`<CalendarContainer/> prop: show should render correctly 1`] = `
<div
  className="calendar__container calendar--show"
>
  <CalendarHead
    onNext={[Function]}
    onPrev={[Function]}
    onTitleClick={[Function]}
    title="2018.12"
  />
  <CalendarBody
    current={"2018-12-04T15:00:00.000Z"}
    locale="en"
    onClick={[Function]}
    viewMode={2}
  />
</div>
`;
